types/violation.go

```diff
@@ -0,0 +1,91 @@
+package types
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// ErrViolations is an error that represent violations to expectations on
+// application configuration.
+type ErrViolations []Violation
+
+var _ error = ErrViolations{}
+
+func (e ErrViolations) Error() string {
+	if len(e) == 1 {
+		return e[0].String()
+	}
+
+	sort.Slice(e, func(i, j int) bool {
+		if c := strings.Compare(e[i].Path, e[j].Path); c != 0 {
+			return c < 0
+		}
+
+		if c := strings.Compare(e[i].SetName, e[j].SetName); c != 0 {
+			return c < 0
+		}
+
+		if c := strings.Compare(e[i].ParamName, e[j].ParamName); c != 0 {
+			return c < 0
+		}
+
+		return e[i].Message < e[j].Message
+	})
+
+	ret := strings.Builder{}
+	ret.WriteString("Multiple violations:\n")
+	for _, violation := range e {
+		vStr := violation.String()
+		ret.WriteString("- " + vStr + "\n")
```

Comment:
optional:

I prefer a more human centered message:
instead of e.g.
```
Multiple violations:
- param="name" value="<missing>" Required
- param="number" value="-123312" strconv.ParseInt: parsing "-123312": value out of range
```

```
Parsing configuration failed:
- "name" parameter is required but was not specified
- "number": invalid value specified: strconv.ParseInt: parsing "-123312": value out of range